buildSrc/src/main/kotlin/configure-compilation-conventions.gradle.kts

```diff
@@ -16,17 +17,26 @@ configure(subprojects) {
             apiVersion = getOverriddenKotlinApiVersion(project)
             if (isMainTaskName && versionsAreNotOverridden) {
                 allWarningsAsErrors = true
+                freeCompilerArgs = freeCompilerArgs + "-Xexplicit-api=strict"
```

Comment:
BTW, as we are here, AFAIK `kotlinOptions` will be deprecated soon-ish and can be replaced already with `compilerOptions` which has API using Gradle providers instead of just getters/setters
Maybe it make sense to replace those usages in this PR? (maybe also in other places, if they exist)
Or even move this configuration from tasks to compilations?